web/client/plugins/TOC/components/ArcGISLegend.jsx

```diff
@@ -17,24 +18,39 @@ import { getLayerIds } from '../../../utils/ArcGISUtils';
 /**
  * ArcGISLegend renders legend from a MapServer or ImageServer service
  * @prop {object} node layer node options
+ * @prop {number} legendWidth width of the legend symbols
+ * @prop {number} legendHeight height of the legend symbols
+ * @prop {object} mapBbox map bounding box
  */
 function ArcGISLegend({
-    node = {}
+    node = {},
+    legendWidth = 12,
+    legendHeight = 12,
+    mapBbox
 }) {
     const [legendData, setLegendData] = useState(null);
     const [error, setError] = useState(false);
-    const legendUrl = node.url ? `${trimEnd(node.url, '/')}/legend` : '';
+    const legendUrl = node.url ? `${trimEnd(node.url, '/')}/${node.enableDynamicLegend ? 'queryLegends' : 'legend'}` : '';
     useEffect(() => {
         if (legendUrl) {
             axios.get(legendUrl, {
-                params: {
+                params: assign({
                     f: 'json'
-                }
+                }, node.enableDynamicLegend ? {
+                    bbox: Object.values(mapBbox.bounds ?? {}).join(',') || '',
```

Comment:
Bounding box from map is not normalized so at lower zoom level the legend is not correctly updated.
In this case we can use the [getExtentFromViewport](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/CoordinatesUtils.js#L749-L759) utils to get a normalized and valid bbox

```js
const extent = getExtentFromViewport(mapBbox, 'EPSG:4326') || [];
axios.get(legendUrl, {
    params: {
        f: 'json',
        ...((node.enableDynamicLegend && isMapServerUrl(node.url)) && {
            bbox: extent.join(','),
            bboxSR: extent.length ? '4326' : '',
```